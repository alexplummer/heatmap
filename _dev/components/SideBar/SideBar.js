
// SideBar
// ============
// Holds further information about the map

// Imports
import React from 'react';
import Stats from '../Stats/Stats';
import VisitTable from '../VisitTable/VisitTable';

class SideBar extends React.Component {

    constructor(props) {
        super(props);

        this.state = {
            active: false,
            locations: [],
        }

        this.sortedLocations = [];
        this.allSortedLocations = [];
        this.winningArray = [];
    }

    sortLocationVisits() {
        this.sortedLocations = [];

        for (let eachLocation in this.props.locations) {

            if (this.props.locations.hasOwnProperty(eachLocation)) {
                this.sortedLocations.push([eachLocation, this.props.locations[eachLocation]]);
            }
        }

        this.sortedLocations.sort((a, b) => {
            return b[1] - a[1];
        });

        this.allSortedLocations = this.sortedLocations.slice();

        this.winningArray = this.sortedLocations.splice(0, 1);
    }

    componentWillUpdate(nextProps) {

        this.sortLocationVisits();

        if (nextProps.locations !== this.props.locations) {

            this.setState({
                locations: this.locationsList,
                active: true
            });
        }
    }

    render() {

        return (
            <div className={this.state.active ? 'SideBar animated bounceInRight' : 'SideBar'}>
                <Stats locations={this.allSortedLocations} />
                <VisitTable winner={this.winningArray} locations={this.sortedLocations} />
            </div>
        );
    }
}

// Exports
export default SideBar;
