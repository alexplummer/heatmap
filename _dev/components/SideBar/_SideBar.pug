
//- SideBar
//- ============
//- Holds further information about the map

//- Includes

mixin SideBar()
    
    .SideBar
