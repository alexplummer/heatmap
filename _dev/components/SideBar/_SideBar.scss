
// SideBar
// ============
// Holds further information about the map

.SideBar {
    position: fixed;
    top: 0;
    right: 0;
    width: 36vw;
    height: 100vh;
    background: #fff;
    z-index: $z-m;
}