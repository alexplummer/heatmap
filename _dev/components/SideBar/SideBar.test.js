
// MainHeader
// ============
// Heading for the document

// Imports
import React from 'react';
import renderer from 'react-test-renderer';
import SideBar from './SideBar';

describe('SideBar renders correctly', () => {
    it('renders correctly', () => {
        const locations = {'location1' : 1, 'location2' : 2};

        const rendered = renderer.create(
            <SideBar locations={locations} />
        );
        expect(rendered.toJSON()).toMatchSnapshot();
    });
});