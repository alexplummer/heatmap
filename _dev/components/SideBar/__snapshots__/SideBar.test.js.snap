// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SideBar renders correctly renders correctly 1`] = `
<div
  className="SideBar"
>
  <div
    className="Stats"
  />
  <div
    className="VisitTable"
  >
    <h3>
      Current winner
    </h3>
    <table
      className=""
    >
      <tr>
        <th>
          Name
        </th>
        <th>
          Visits
        </th>
      </tr>
    </table>
    
    <table />
  </div>
</div>
`;
